### PR TITLE
Replace Gradle with Maven

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

The main motivation is that with Gradle we need to update it each time
there is a new JDK version and Gradle 7.5 breaks our setup in a way that
you cannot load the project anymore with Eclipse/VSCode/eclipse.jdt.ls.

The report upstream got no attention and meanwhile they released Gradle
7.6 and now 8.0.

Gradle 8.0 contains some further changes that would break our build
without adjustments.

Currently Gradle also doesn't execute the testFixtures test-cases, which is
something we'd need to fix.

## TODO

- [x] Get tests to pass (There's some weird thing with the antlr grammar generation where `SqlBaseLexer.VOCABULARY` is now missing)
    - [ ] ~Fix test cases in testFixtures which are broken (not executed by gradle. E.g. `testAddDuplicateTableThrowsException` in `SQLExecutorTest`)~ Deleted for now because other tests already depend on this behavior. We can re-add this later if desired.
- [x] Add pom.xml for mandatory plugins
- [x] Create assembly
- [x] Add pom.xml for runtime plugins/modules
- [x] Include ⬆️ in assembly
- [x] Bundle AdminUI
- [x] Add blackbox build
- [x] Add checkstyle
- [x] Adapt CI configuration
- [x] Adapt readme/docs
- [ ] ~Follow up on packaging/JDK bundling (jpackage, jlink)~

Might still be missing some adaptions. But should be good enough for a review now

### CODE REVIEW

**Comment by matriv** [comment]
:fearful: :scream:

**Comment by robd003** [comment]
Just curious @mfussenegger, what exactly broke that made it so you couldn't load the project in IDEs?

**Comment by mfussenegger** [comment]
> Just curious @mfussenegger, what exactly broke that made it so you couldn't load the project in IDEs?

It will load but cannot build the full project anymore. Something about the server module not being able to build before it's dependencies have been built. I suspect it's related to the test fixtures (which we migrated to, after a previous gradle update broke the previous solution ...). Unfortunately there's not much more info going on. I tried to debug a bit to get more insights but I don't really have a picture of how buildship/gradle/rest-of-eclipse interacts, and who is responsible for what.

File: libs/pgwire/src/main/antlr/io/crate/protocols/postgres/antlr/PgArray.g4 [hunk 1]
```diff
@@ -21,6 +21,7 @@
 
 grammar PgArray;
 
+
```
Comment (mkleen) [inline libs/pgwire/src/main/antlr/io/crate/protocols/postgres/antlr/PgArray.g4:24]: Minor: That was probably unneeded.

**Review by mkleen — APPROVED** [review]
Great work!

**Pull request merged**